definitions/checks/repositories/check_upstream_repository.rb

```diff
@@ -0,0 +1,65 @@
+class Checks::CheckUpstreamRepository < ForemanMaintain::Check
+  metadata do
+    label :check_upstream_repository
+    description 'Check if any upstream repositories are enabled on system'
+    tags :pre_upgrade
+    preparation_steps do
+      Procedures::Packages::Install.new(:packages => %w[yum-utils])
+    end
+    confine do
+      feature(:downstream)
+    end
+  end
+
+  def run
+    enabled_upstream_repos = []
+    with_spinner('Checking for presence of upstream repositories') do
+      run_compare(enabled_upstream_repos)
+    end
+    repos_to_disable = repo_label(enabled_upstream_repos)
+    next_steps = []
+    next_steps.concat(
+      [Procedures::Repositories::Disable.new(:repos => repos_to_disable)]
+    )
+    assert(enabled_upstream_repos.empty?,
+           "System has upstream #{repos_to_disable} repositories enabled",
+           :next_steps => next_steps)
+  end
+
+  def repo_label(repos)
```

Comment:
As this method returns comma separated list of repositories labels, please modify name of it.
e.g. `comma_separated_repositories` or `comma_separated_repositories_labels`


Also, you can modify code to `repos.map { |r| r.gsub(%r{Repo-id:|\/+\w*}, '') }.join(',')`